src/protocol/ldap/org/apache/jmeter/protocol/ldap/sampler/LdapExtClient.java

```diff
@@ -78,14 +78,19 @@ public LdapExtClient() {
      * @exception NamingException
      *                when creating the {@link DirContext} fails
      */
-    public static DirContext connect(String host, String port, String rootdn, String username, String password, String connTimeOut, boolean secure)
+    public static DirContext connect(String host, String port, String rootdn, String username, String password, String connTimeOut, 
+            boolean secure, boolean trustAll)
             throws NamingException {
         DirContext dirContext;
         Hashtable<String, String> env = new Hashtable<>();
         env.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.ldap.LdapCtxFactory"); // $NON-NLS-1$
         StringBuilder sb = new StringBuilder(80);
         if (secure) {
             sb.append("ldaps://"); // $NON-NLS-1$
+            if (trustAll){
+                log.debug("Secure true, trustAll true");
+                env.put("java.naming.ldap.factory.socket", "org.apache.jmeter.protocol.ldap.sampler.TrustAllSocket");
```

Comment:
My initial thought was to add a parameter that would have been inserted here, but maybe the switch for `trustAll` is a bit more user friendly.

Reply:
Yeah I went with that because I wanted the user to be able to pick whether they wanted to use the new feature or not.